### PR TITLE
Downstream Core tests on SciMLSensitivity

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
These almost all hit Enzyme and stress test it. Core2 fails on latest release, this should test master.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Bumping this. https://github.com/SciML/SciMLSensitivity.jl/pull/763 could've been prevented with the bigger test suite!

File: .github/workflows/Downstream.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+jobs:
+  test:
+    name: ${{ matrix.package.repo }}/${{ matrix.package.group }}/${{ matrix.julia-version }}
+    runs-on: ${{ matrix.os }}
+    env:
+      GROUP: ${{ matrix.package.group }}
+    strategy:
+      fail-fast: false
+      matrix:
+        julia-version: [1,1.6]
```
Comment (vchuravy) [inline .github/workflows/Downstream.yml:17]: Only test `1.6`?

Comment (ChrisRackauckas) [reply .github/workflows/Downstream.yml:17]: only v1 is probably more important than only v1.6, if the problem is number of tests.

**Comment by wsmoses** [comment]
@ChrisRackauckas FYI on main we now have a slightly different calling convention as we prepare for 0.11.

**Comment by ChrisRackauckas** [comment]
I couldn't tell from the docs. Are the dev version of the docs just not updated yet?

**Comment by wsmoses** [comment]
They should be now, reload

**Comment by ChrisRackauckas** [comment]
I still don't see what's different. Are there any notes that describes what changed?

**Comment by wsmoses** [comment]
Mode is now a required argument, and non-active arguments in reverse mode will return a nothing in the tuple (such that the n-th element of the tuple corresponds to the n-th argument). Finally the derivative tuple in reverse mode is now wrapped in a second tuple (e.g. (derivative vals, ) ). This enables the new API for also having reverse mode return the original return of the function as well as the second tuple element.

https://enzymead.github.io/Enzyme.jl/dev/api/#Enzyme.autodiff-Union{Tuple{ReturnPrimal},%20Tuple{A},%20Tuple{F},%20Tuple{EnzymeCore.ReverseMode{ReturnPrimal},%20F,%20Type{A},%20Vararg{Any}}}%20where%20{F,%20A%3C:EnzymeCore.Annotation,%20ReturnPrimal}

**Comment by vchuravy** [comment]
In general it would be great if we could identify a few relevant test to include as part of our CI suite, I am hesitant to take on the additional CI time.

**Comment by ChrisRackauckas** [comment]
Paying for CI time is a lot cheaper than human time.

**Pull request closed without merging**